### PR TITLE
feat!(types): Add Blob::index field introduced in celestia 0.13

### PR DESCRIPTION
**Pull request by S1nus**
Another attempt at adding this field. Now, trying to use an `Option<u64>`.

### CODE REVIEW

File: types/src/blob.rs [hunk 1]
```diff
@@ -52,6 +52,20 @@ pub struct Blob {
...
     pub commitment: Commitment,
+    #[serde(deserialize_with = "deserialize_optional_positive_integer")]
+    pub index: Option<u64>,
+}
+
+fn deserialize_optional_positive_integer<'de, D>(deserializer: D) -> Result<Option<u64>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let value: i64 = Deserialize::deserialize(deserializer)?;
```
Comment (zvolin) [inline types/src/blob.rs:63]: nit

```suggestion
    let value = i64::deserialize(deserializer)?;
```

File: types/src/blob.rs [hunk 2]
```diff
@@ -52,6 +52,20 @@ pub struct Blob {
     pub share_version: u8,
     /// A [`Commitment`] computed from the [`Blob`]s data.
     pub commitment: Commitment,
+    #[serde(deserialize_with = "deserialize_optional_positive_integer")]
+    pub index: Option<u64>,
+}
+
+fn deserialize_optional_positive_integer<'de, D>(deserializer: D) -> Result<Option<u64>, D::Error>
```
Comment (zvolin) [inline types/src/blob.rs:59]: nit, could you move it to the very bottom just before `mod tests`?

File: types/src/blob.rs [hunk 3]
```diff
@@ -52,6 +52,20 @@ pub struct Blob {
     pub share_version: u8,
     /// A [`Commitment`] computed from the [`Blob`]s data.
     pub commitment: Commitment,
+    #[serde(deserialize_with = "deserialize_optional_positive_integer")]
```
Comment (zvolin) [inline types/src/blob.rs:55]:
```suggestion
    /// Index of the blob's first share in the EDS. Only set for blobs retrieved from chain.
    #[serde(deserialize_with = "deserialize_optional_positive_integer")]
```

**Review by zvolin — APPROVED** [review]
Looks good :+1: left some suggestions

**Comment by zvolin** [comment]
could you look into the CI failures too?

**Comment by zvolin** [comment]
some of the failures seems to be coming from toolchain updates, I'll look into that tomorrow

File: types/src/blob.rs (hunk 1 above)
Comment (zvolin) [reply types/src/blob.rs]: @S1nus this one seems to be overwritten by later commits. I can't spot this change in diff

**Review by fl0rek — APPROVED** [review]
looks good, thanks for the contribution! 🎸

**Comment by fl0rek** [comment]
@S1nus I've just noticed that the commits aren't signed
Would you mind [setting up commit signing](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) and force-pushing? Checks won't allow merge unless all commits are signed.

**Comment by S1nus** [comment]
@fl0rek I set up signing, newer commits are signed. is it good? or need to do something else?

**Comment by zvolin** [comment]
Previous commits needed to be signed too, but I just squashed them and it looks fine. I've added the fix for celestia-rpc tests, hopefully CI will be happy now

**Review by fl0rek — APPROVED** [review]

**Pull request merged**